src/components/org/apache/jmeter/control/WeightedDistributionController.java

```diff
@@ -0,0 +1,472 @@
+     * 
+     * @see org.apache.jmeter.control.InterleaveControl#incrementCurrent()
+     */
+    @Override
+    protected void incrementCurrent() {
+        super.incrementCurrent();
+        current = determineCurrentTestElement();
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see
+     * org.apache.jmeter.control.GenericController#addTestElement(org.apache.
+     * jmeter.testelement.TestElement)
+     */
+    @Override
+    public void addTestElement(TestElement child) {
+        // if you do not filter enabled elements, if no elements are found by
+        // determineCurrentTestElement() it will return the first element
+        if (child.isEnabled()
+                && child.getPropertyAsInt(WEIGHT, DFLT_WEIGHT) > 0) {
+            super.addTestElement(child);
+        }
+
+    }
+
+    /**
+     * Gets the seed.
+     *
+     * @return the seed
+     */
+    public long getGeneratorSeed() {
+        return getPropertyAsLong(GENERATOR_SEED,
+                WeightedDistributionController.DFLT_GENERATOR_SEED);
+    }
+
+    /**
+     * Sets the seed.
+     *
+     * @param seed
+     *            the new seed
+     */
+    public void setGeneratorSeed(long seed) {
+        if (getGeneratorSeed() != seed) {
+            setProperty(new LongProperty(GENERATOR_SEED, seed));
+            integerGenerator = null;
+        }
+    }
+
+    /**
+     * Gets the randomizer.
+     *
+     * @return the randomizer
+     */
+    public IntegerGenerator getIntegerGenerator() {
+        if (integerGenerator == null) {
+            initIntegerGenerator();
+        }
+        return integerGenerator;
+    }
+
+    /**
+     * Sets the randomizer.
+     *
+     * @param intgen
+     *            the new randomizer
+     */
+    public void setIntegerGenerator(IntegerGenerator intgen) {
+        integerGenerator = intgen;
+    }
+
+    /**
+     * Gets the child node.
+     *
+     * @param idx
+     *            the index of the child node
+     * @return the child node
+     */
+    public JMeterTreeNode getChildNode(int idx) {
+        JMeterTreeNode childNode;
+
+        try {
+            childNode = (JMeterTreeNode) getNode().getChildAt(idx);
+        } catch (Exception ex) {
+            log.error("Unable to retreive child node at index: " + idx, ex);
+            childNode = null;
+        }
+
+        return childNode;
+    }
+
+    /**
+     * Gets the child test element.
+     *
+     * @param idx
+     *            the index of the child node
+     * @return the child test element
+     */
+    public TestElement getChildTestElement(int idx) {
+        JMeterTreeNode childNode = getChildNode(idx);
+        return childNode == null ? null : childNode.getTestElement();
+    }
+
+    /**
+     * Gets the cumulative probability.
+     *
+     * @return the cumulative probability
+     */
+    public int getCumulativeProbability() {
+        // recalculate if reset flag is set
+        if (resetCumulativeProbability) {
```

Comment:
Can you clarify when weight is recalculated?
The documentation suggests that weight is recalculated at run-time, however the flag `resetCumulativeProbability` is reset at design time only. That is it looks like weight distribution would not account for the variable changes in the runtime. Is that the case? Can you make documentation/intention more clear?